File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.py
from collections import deque

def solution(players, callings):
    aDic,bDic={},{}
    for i in range(len(players)):
        p=players[i]
        aDic[i]=p
        bDic[p]=i

    

    for player in callings:
        a=bDic[player]  ## 추월 선수 (player,a)
        p=aDic[a-1] ## 추월 당하는 선수 (p, a-1)
        bDic[player]=a-1
        bDic[p]=a
        aDic[a-1]=player
        aDic[a]=p
    answer = []
    for i in range(len(players)):
        answer.append(aDic[i])
    

        
    return answer

File: 프로그래머스/1/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    const p = {}
    const order = {}
    for(let i=0;i<players.length;i++){
        p[players[i]]=i
        order[i]=players[i]
    }
    for(let name of callings){
        const prev_order = p[name]-1
        const prev_name = order[prev_order]
        order[prev_order]=name
        order[p[name]]=prev_name
        p[name]-=1
        p[prev_name]+=1
        //console.log(name)
        //console.log(order,p)
    }
    const answer = []
    for(let i=0;i<players.length;i++){
        answer.push(order[i])
    }
    //console.log(answer)

    return answer;
}